pkg/volume/configmap/configmap.go

```diff
@@ -192,26 +192,35 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return nil
 }
 
-func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
-	payload := make(map[string][]byte, len(configMap.Data))
+func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap, defaultMode *int32) (payload map[string][]byte, mode map[string]int32, err error) {
```

Comment:
On Tue, Aug 02, 2016 at 08:53:29PM -0700, Tim Hockin wrote:
> > @@ -192,26 +192,35 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
> >  	return nil
> >  }
> >  
> > -func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
> > -	payload := make(map[string][]byte, len(configMap.Data))
> > +func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap, defaultMode *int32) (payload map[string][]byte, mode map[string]int32, err error) {
> 
> what happened to the idea of a map[string]struct?  I really dislike parallel maps.

I plan to do it, but wasn't sure if @pmorie said instead of "projection"
something called some other way. That's why I didn't ping you, because this
wasn't ready =)

But yeah, I totally agree that something like that is needed and way better. I
just did some basic changes yesterday. More should come :-)